### PR TITLE
feat(ui, tests): Add Seeker and Provider registration screens

### PR DESCRIPTION
**Pull request by rimabk**
This PR introduces the following changes:

**New Registration Screens:**
    **Seeker Registration Screen:**
       Implemented the screen for seekers where they can input their information.
       The registration process consists of three steps:
      **Step 1:** Enter personal information.
      **Step 2:** Choose preferences (this step will be implemented later).
      **Step 3:** Completion screen showing "All done!"
      **Provider Registration Screen:**
      A similar three-step process for providers to input their details and preferences.
**UI Tests:**
   Added UI tests for both the SeekerRegistrationScreen and ProviderRegistrationScreen.
   Ensured that all UI elements are properly displayed and functional across the registration steps.

**Improved Unit Tests:**
  **SeekerViewModel:**
     Added more unit tests to cover additional scenarios.
   **UserRepository:**
    Enhanced the existing test cases to cover more edge cases and improve reliability.
    Deleted unused functions to improve code cleanliness.

### CODE REVIEW

**Review by manunam2002 — APPROVED** [review]
### PR Review

#### General Feedback
1. **Code Quality:**
The code is well-structured and follows Kotlin conventions. The work that you have done on the UI is impressive!

2. **Tests:**
Tests are exhaustive and reach a great line coverage, which is excellent.

3. **Improvements:**
Consider optimizing any repeated operations, there is some repeated code in the UI part, consider modularisation for the future.

4. **Conclusion**
Overall, great addition to the project, good job!

File: app/src/main/java/com/android/solvit/seeker/model/profile/SeekerProfileViewModel.kt
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/seeker/model/profile/SeekerProfileViewModel.kt:1]: seeker profile should be a state flow of a single instance of seeker profile, not a list of them

**Review by hjamaly — CHANGES_REQUESTED** [review]
**Feedback** 
This PR implements an important functionnality (sign up) with more than 6 UI's implemented. The work is impressive and UI are well designed

**Code**
Code is clear and understandable and formatted

**To Change**
You tried to implement some functionalities that we already implemented in the backend
(add User Profile or seekerProfile) . Seeker profile for example should be a state flow of a single instance and not a list. I suggest to you to pull the main and remerge your branch with it and everything will be good ! Also  User Repository Firestore has not to be an open class

File: app/src/main/java/com/android/solvit/seeker/model/profile/UserRepositoryFirestore.kt
Comment (manunam2002) [inline app/src/main/java/com/android/solvit/seeker/model/profile/UserRepositoryFirestore.kt:1]: collectionPath should be seeker for consistency

Comment (manunam2002) [reply app/src/main/java/com/android/solvit/seeker/model/profile/UserRepositoryFirestore.kt:1]: small detail: parameter in deleteUserProfile should be uid for consistency

**Review by hjamaly — APPROVED** [review]
Since the collaborator encountered some technical issues and was unable to update their branch, I have made the necessary updates. These changes reflect the requested modifications, and the branch is now up to date.

**Pull request merged**